File: src/network/packet/impl/request/client/CommandExecuteRequestPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\request\client;

use pocketcloud\cloud\network\packet\RequestClientPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class CommandExecuteRequestPacket extends RequestClientPacket {

    public function __construct(
        private readonly string $commandLine = "",
        private readonly string $id = ""
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->commandLine, $this->id);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getCommandLine(): string {
        return $this->commandLine;
    }

    public function getId(): string {
        return $this->id;
    }

    public static function create(string $commandLine, string $id): self {
        return new self($commandLine, $id);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/request/client/CommandExecuteRequestPacket.java
package de.pocketcloud.cloud.network.packet.impl.request.client;

import de.pocketcloud.cloud.network.packet.RequestClientPacket;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class CommandExecuteRequestPacket extends RequestClientPacket {

    private String commandLine;
    private String id;

    public CommandExecuteRequestPacket(String commandLine, String id) {
        this.commandLine = commandLine != null ? commandLine : "";
        this.id = id != null ? id : "";
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(commandLine, id);
    }

    public static CommandExecuteRequestPacket create(String commandLine, String id) {
        return new CommandExecuteRequestPacket(commandLine, id);
    }
}
